src/classes/PS_ElevateService.cls

```diff
@@ -0,0 +1,94 @@
+/*
+    Copyright (c) 2020, Salesforce.org
+    All rights reserved.
+
+    Redistribution and use in source and binary forms, with or without
+    modification, are permitted provided that the following conditions are met:
+
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+      
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2020
+* @group Elevate Payment Services
+* @description Contains Elevate payment services methods 
+*/
+public without sharing class PS_ElevateService {
+
+    /***
+    * @description Checks whether an org is a customer of Elevate Payment Services
+    * @return Boolean
+    */
+    @AuraEnabled
+    public static Boolean isElevateCustomer() {
```

Comment:
Is this method used anywhere?

Reply:
It is not used currently (it did at some part of the RD2 credit widget but decided to use `RD2_ElevateIntegrationService.isIntegrationEnabled()`), but most importantly it was created so Gift Entry code referencing common Elevate methods can start using the `PS_ElevateService`. Should I remove it for now, and leave this to Honeygems to add when they refactor the code to use the common methods?
CC @ptewson-sfdo

Reply:
Thanks for the clarification. There are other parts of the code that return back the elevate status as part of wrapper class structure. Wondering if that's the pattern to stick with even for future code.

I suppose, along the same lines, the `getOrgDomainInfo()` method is more of a utility method than something  specific to PaymentServices, though it is being used specifically by PS right now. Rather than include the AuraEnabled method in this class, which would then require permissions granted, could there be a `UTIL_LWC` (or `LWC_Utilities') class that has exposed utility methods, even if those methods delegate to other service classes? That could also help with permissions going forward.

Reply:
I understand for `getOrgDomainInfo()` that is a generic even though it is not used outside of the Elevate wrapper currently, but I assume we will have many Elevate related AuraEnabled methods so we do need a class specific to Elevate. I would avoid mixing Elevate AuraEnabled and generic methods in one class.

Reply:
That's exactly my concern here. `getOrgDomainInfo` is generic and is in this class, while this class is meant to be specific to PaymentServices related methods.

Reply:
That noted, if this class is going to expose AuraEnabled methods we can leave `getOrgDomainInfo` in here for now and eventually move out. The benefit of moving now though is that we can add the new utility class permission to our unpackaged metadata config for new RD2 users.

Reply:
The new `UTIL_AuraEnabledCommon` has been created a generic utility class for common methods regardless of the LC/LWC domain. Its name and/or `PS_ElevateService` can be changed before merging into `master`, and we will discuss this with Honeygems next week at the dev meeting. CC @ptewson-sfdo